crates/ide/src/highlight_related.rs

```diff
@@ -232,6 +232,22 @@ fn highlight_references(
             }
         }
 
+        if matches!(def, Definition::Label(_)) {
```

Comment:
We should start commenting this file a bit more (I am partially at fault for not doing so here :)

So lets put a comment above this stating what this is doing, highlighting the tail expr of labelled blocks